WordPress/Classes/ViewRelated/Plugins/PluginDirectoryViewModel.swift

```diff
@@ -12,36 +12,71 @@ class PluginDirectoryViewModel: Observable {
 
     let changeDispatcher = Dispatcher<Void>()
 
+    private(set) var refreshing = false {
+        didSet {
+            if refreshing != oldValue {
+                emitChange()
+            }
+        }
+    }
     private let store: PluginStore
-    private var storeReceipt: Receipt?
 
-    private var installedReceipt: Receipt?
-    private var featuredReceipt: Receipt?
-    private var popularReceipt: Receipt?
-    private var newReceipt: Receipt?
+    private let installedReceipt: Receipt
+    private let featuredReceipt: Receipt
+    private let popularReceipt: Receipt
+    private let newReceipt: Receipt
+
+    private var storeReceipt: Receipt?
+    private var actionReceipt: Receipt?
 
     public init(site: JetpackSiteRef, store: PluginStore = StoreContainer.shared.plugin) {
         self.store = store
         self.site = site
 
         installedReceipt = store.query(.all(site: site))
-        featuredReceipt = store.query(.featured(site: site))
+        featuredReceipt = store.query(.featured)
         popularReceipt = store.query(.feed(type: .popular))
         newReceipt = store.query(.feed(type: .newest))
 
         storeReceipt = store.onChange { [weak self] in
             self?.changeDispatcher.dispatch()
         }
+
+        actionReceipt = ActionDispatcher.global.subscribe { [weak self] (action) in
+            self?.refreshRefreshing()
```

Comment:
This should happen on `store.onChange`. Nothing should probably subscribe to ActionDispatcher unless it's going to process the given Action